synapse/lib/spooled.py

```diff
@@ -60,6 +60,16 @@ async def __anit__(self, dirn=None, size=10000, cell=None):
         self.realset = set()
         self.len = 0
 
+    async def __aiter__(self):
+
+        if not self.fallback:
+            for item in self.realset:
```

Comment:
We'll have to be careful with this construct in the future.
Before we fallback, if we add items during iteration that will raise a RuntimeError. In fallback mode the slab scan does not raise an error.